content/blog/ggplot-3-5-0-guides/index.Rmd

```diff
@@ -0,0 +1,887 @@
+  guides(
+    colour = guide_colourbar(position = "bottom"),
+    size   = guide_legend(position = "top"),
+    alpha  = guide_legend(position = "inside")
+  ) +
+  theme(legend.position = "left")
+p
+```
+
+In the plot above, the legend for the 'cyl' variable is in the middle of the plot.
+In previous versions of ggplot2, you could set the `legend.position` to a coordinate to control the placement.
+However, doing this would change the default legend position, which is not always desirable.
+To cover such cases, there is now a specialised `legend.position.inside` argument that controls the positioning of legends with `position = "inside"` regardless of whether the position was specified in the theme or in the guide.
+
+```{r legend_inside}
+#| fig.alt = "The same plot as before, but the legend for the 'cyl' variable is
+#|  to the top-right of the centre."
+
+p + theme(legend.position.inside = c(0.7, 0.7))
+```
+
+The justification of legends is controllable by using the `legend.justification.{position}` theme setting.
+Moreover, the top and bottom guides can be aligned to the plot rather than the panel by setting the `legend.location` argument.
+The main reason behind this is that you can then align the legends with the plot's title.
+By default, when `plot.title.position = "plot"`, left legends are already aligned. 
+For this reason, the top and bottom guides are prioritised for the `legend.location` setting.
+Moreover, it avoids overlapping of legends in the corners if the justifications would dictate it.
+
+```{r legend_alignments}
+#| fig.alt = "The same plot as before, but with a plot-aligned title and 
+#|  different alignments of the legends. The left and top legends are 
+#|  left-aligned with the title."
+
+p + 
+  labs(title = "Plot-aligned title") +
+  theme(
+    legend.margin = margin(0, 0, 0, 0), # turned off for alignment
+    legend.justification.top = "left",
+    legend.justification.left = "top",
+    legend.justification.bottom = "right",
+    legend.justification.inside = c(1, 1),
+    legend.location = "plot",
+    plot.title.position = "plot"
+  )
+```
+
+### Awareness
+
+Legends are now more aware what discrete variables should be placed in which keys.
+By default, they now only draw keys for the layer which contain the relevant value.
+This saves one having to hassle with the `guide_legend(override.aes)` argument to get the keys to display just right.
+In the plot below, notice how the points and line have separate keys.
+
+```{r legend_awareness}
+#| fig.alt = "A scatterplot with trendline showing engine displacement versus
+#|  highway miles per gallon. There are two legends for colour and alpha. Both
+#|  legends show points and lines separately."
+
+p <- ggplot(mpg, aes(displ, hwy)) +
+  scale_alpha_manual(values = c(0.5, 1))
+p +
+  geom_point(aes(colour = "points", alpha = "points")) +
+  geom_line(
+    aes(colour = "line", alpha = "line"),
+    stat = "smooth", formula = y ~ x, method = "lm"
+  )
```

Comment:
I think this would look better if you merged the two legends.

Reply:
I agree about the looks, but I wanted them separated here for continuity with next plot.